embassy-rp/src/pwm.rs

```diff
@@ -344,6 +344,62 @@ impl<'d> Pwm<'d> {
     fn bit(&self) -> u32 {
         1 << self.slice as usize
     }
+
+    #[inline]
+    /// Split Pwm driver to allow separate duty cycle control of each channel
+    pub fn split(&self) -> (PwmOutput,PwmOutput){
+        (PwmOutput::new(PwmChannel::A,self.slice.clone()),PwmOutput::new(PwmChannel::B,self.slice.clone()))
+    }
+}
+
+enum PwmChannel{
+    A,
+    B
+}
+
+/// Single channel of Pwm driver.
+pub struct PwmOutput {
```

Comment:
this should have a `'d` lifetime same as `Pwm`, so you can't make it outlive the peripheral singletons when you create it with borrowed singletons, like `Pwm::new(&mut p.PWM_SLICE3, ...)`.